source/components/molecules/ImageUploader/ImageUploader.tsx

```diff
@@ -0,0 +1,278 @@
+/* eslint-disable no-nested-ternary */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { NativeSyntheticEvent, NativeScrollEvent, TouchableOpacity } from 'react-native';
+import ImagePicker, { Image as CropPickerImage } from 'react-native-image-crop-picker';
+import styled from 'styled-components/native';
+import { Text, Button, Icon, Label } from '../../atoms';
+import { Modal, useModal } from '../Modal';
+import uploadFile, { getBlob } from '../../../helpers/FileUpload';
+import HorizontalScrollIndicator from '../../atoms/HorizontalScrollIndicator';
+import { getValidColorSchema, PrimaryColor } from '../../../styles/themeHelpers';
+import ImageItem from './ImageItem';
+
+const Wrapper = styled.View`
+  padding-left: 0;
+  padding-right: 0;
+  padding-top: 15px;
+  padding-bottom: 0;
+`;
+const Container = styled.ScrollView`
+  padding-left: 16px;
+  padding-right: 16px;
+`;
+const ButtonContainer = styled.View`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  margin-top: 25px;
+  width: 100%;
+`;
+const BackgroundBlur = styled.View`
+  position: absolute;
+  z-index: 1000;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  padding: 0px;
+  background-color: rgba(0, 0, 0, 0.25);
+`;
+
+const PopupContainer = styled.View<{colorSchema: PrimaryColor}>`
+  position: absolute;
+  z-index: 1000;
+  top: 60%;
+  left: 5%;
+  right: 5%;
+  padding: 20px;
+  width: 90%;
+  background-color: ${(props) => props.theme.colors.complementary[props.colorSchema][0]}}
+  border-radius: 6px;
+  shadow-offset: 0 0;
+  shadow-opacity: 0.1;
+  shadow-radius: 6px;
+  justify-content: space-between;
+`;
+
+const Row = styled.View`
+  flex-direction: row;
+  justify-content: space-between;
+`;
+
+const PopupLabel = styled(Label)<{colorSchema: PrimaryColor}>`
+  color: ${props => props.theme.colors.primary[props.colorSchema][0]}
+`;
+
+const PopupButton = styled(Button)`
+  border: 0;
+  margin-bottom: 16px;
+`;
+
+interface Image extends CropPickerImage {
+  errorMessage?: string;
+  uploadedFileName?: string;
+  url?: string;
+}
+
+export type ImageStatus = 'loading' | 'uploaded' | 'error';
+
+interface Props {
+  buttonText: string;
+  value: Image[];
+  onChange: (value: Record<string, any>[]) => void;
+  colorSchema?: PrimaryColor;
+  maxImages?: number;
+}
+
+const ImageUploader: React.FC<Props> = ({ buttonText, value: oldImages, onChange, colorSchema, maxImages }) => {
+  const [images, setImages] = useState<Image[]>([]);
+  const [loadedStatus, setLoadedStatus] = useState<ImageStatus[]>([]);
+  const [horizontalScrollPercentage, setHorizontalScrollPercentage] = useState(0);
+  const [choiceModalVisible, toggleModal] = useModal();
+
+  useEffect(() => {
+    // if the component is sent a list of existing images, we load them into the state
```

Comment:
Remove comment, the if statement is clear enough to understand without it :)

Reply:
Done